src/System.Runtime.Extensions/src/System/IO/PathHelper.Windows.cs

```diff
@@ -278,19 +234,20 @@ private string TryExpandShortFileName(StringBuilder outputBuffer)
             Debug.Assert(!PathInternal.IsExtended(outputBuffer), "expanding short names expects normal paths");
 
             // Add the extended prefix before expanding to allow growth over MAX_PATH
-            char[] inputBuffer = null;
-            int rootLength = PathInternal.GetRootLength(outputBuffer);
-            bool isUnc = this.IsUnc(outputBuffer);
-            int rootDifference = this.GetInputBuffer(outputBuffer, isUnc, out inputBuffer);
+            StringBuffer inputBuffer = null;
+            ulong rootLength = PathInternal.GetRootLength(outputBuffer);
+            bool isUnc = IsUnc(outputBuffer);
+
+            ulong rootDifference = GetInputBuffer(outputBuffer, isUnc, out inputBuffer);
             rootLength += rootDifference;
-            int inputLength = inputBuffer.Length;
+            ulong inputLength = inputBuffer.Length;
 
             bool success = false;
-            int foundIndex = inputBuffer.Length - 1;
+            ulong foundIndex = inputBuffer.Length - 1;
 
             while (!success)
             {
-                int result = Interop.mincore.GetLongPathNameW(inputBuffer, outputBuffer, outputBuffer.Capacity + 1);
+                uint result = Interop.mincore.GetLongPathNameW(inputBuffer.GetHandle(), outputBuffer.GetHandle(), (uint)outputBuffer.CharCapacity);
```

Comment:
Possibly- I've left that for future investigation. I'm not sure what the fail case does- I need to extract information out of the input when it fails- if it gets munged I can't leverage this. I'll create an issue to track it.